src/utils.ts

```diff
@@ -494,47 +494,44 @@ function parseHeadStream<T extends JSONRPCMessage>(
   const endP = promise();
   parser.onEnd = () => endP.resolveP();
 
-  return new TransformStream<Uint8Array, T | Uint8Array>(
-    {
-      flush: async () => {
-        if (!parser.isEnded) parser.end();
-        await endP.p;
-      },
-      start: (controller) => {
-        parser.onValue = async (value) => {
-          const jsonMessage = messageParser(value.value);
-          controller.enqueue(jsonMessage);
-          bytesWritten = 0;
-          parsing = false;
-        };
-      },
-      transform: async (chunk, controller) => {
-        if (parsing) {
-          try {
-            bytesWritten += chunk.byteLength;
-            parser.write(chunk);
-          } catch (e) {
-            throw new errors.ErrorRPCParse(undefined, {
-              cause: e,
-            });
-          }
-          if (bytesWritten > bufferByteLimit) {
-            throw new errors.ErrorRPCMessageLength();
-          }
-        } else {
-          // Wait for parser to end
-          if (!ended) {
-            parser.end();
-            await endP.p;
-            ended = true;
-          }
-          // Pass through normal chunks
-          controller.enqueue(chunk);
+  return new TransformStream<Uint8Array, T | Uint8Array>({
+    flush: async () => {
+      if (!parser.isEnded) parser.end();
+      await endP.p;
+    },
+    start: (controller) => {
+      parser.onValue = async (value) => {
+        const jsonMessage = messageParser(value.value);
+        controller.enqueue(jsonMessage);
+        bytesWritten = 0;
+        parsing = false;
+      };
+    },
+    transform: async (chunk, controller) => {
+      if (parsing) {
+        try {
+          bytesWritten += chunk.byteLength;
+          parser.write(chunk);
+        } catch (e) {
+          throw new errors.ErrorRPCParse(undefined, {
+            cause: e,
+          });
         }
-      },
+        if (bytesWritten > bufferByteLimit) {
+          throw new errors.ErrorRPCMessageLength();
+        }
+      } else {
+        // Wait for parser to end
+        if (!ended) {
+          parser.end();
+          await endP.p;
+          ended = true;
+        }
+        // Pass through normal chunks
+        controller.enqueue(chunk);
+      }
     },
-    { highWaterMark: 1 },
-  );
```

Comment:
Don't remove the `highWaterMark`. We need all streams being used to only hold 1 message at a time.